R/simulation-set.R

```diff
@@ -25,22 +28,29 @@ SimulationSet <- R6::R6Class(
     #' @param observedDataFile name of csv file to be used for observed data
     #' @param observedMetaDataFile name of csv file to be used as dictionary of the observed data
     #' @param timeUnit display unit for time variable. Default is "h"
+    #' @param applicationRanges named list of logicals defining which application ranges are included in
+    #' time profiles and residual plots when applicable. Names are available in enum `ApplicationRanges`.
+    #' Providing directly a logical value will associate the value to all application ranges.
     #' @return A new `SimulationSet` object
     initialize = function(simulationSetName,
                               simulationFile,
                               outputs = NULL,
                               observedDataFile = NULL,
                               observedMetaDataFile = NULL,
-                              timeUnit = "h") {
+                              timeUnit = "h",
+                              applicationRanges = TRUE) {

```

Comment:
`applicationRanges = TRUE` ??
I expected the structure with 3 fields (first, last, total) (each of them TRUE per default)?

Reply:
The input also accepts a named list with firstApplication, lastApplication and total.
If the user input `TRUE`/`FALSE` directly all 3 values become true/false.
If it is better for users, I can replace `applicationRanges` by 3 separate inputs  firstApplication, lastApplication and total.

Reply:
I would in fact suggest to provide the 3-list only.
In the current setup, TRUE is the default and can be omitted anyway, FALSE does not really make sense (why would you define time profile task without any time profiles) - thus if users want to override the setting, they will always provide a 3-list.